### PR TITLE
PLY: Support writing per-face/vertex property lists

### PR DESCRIPTION
**Pull request by nh2**
## Summary of Changes

Hi, I'd like to contribute the writing of PLY property lists.

This allows to directly write multi-texture PLY files with CGAL that can be opened directly in e.g. MeshLab, as well as other use cases that require `property list`.

## Release Management

* Affected package(s):
* Issue(s) solved (if any): fix #0000, fix #0000,...
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

File: Stream_support/include/CGAL/IO/PLY/PLY_writer.h [hunk 1]
```diff
@@ -298,6 +298,42 @@ class Char_property_printer
...
+    {
+      stream << vec.size();
+      for(const ElementType& v : vec)
+      {
+        stream << " " << v;
+      }
+    }
+    else
+    {
+      unsigned char size = (unsigned char)(vec.size());
```
Comment (nh2) [inline Stream_support/include/CGAL/IO/PLY/PLY_writer.h:326]: I don't like this potentially-overflowing cast.

I copied it from existing code that uses the same logic:

https://github.com/CGAL/cgal/blob/acbd1ee182211c6309f355925927802f036fbe13/Surface_mesh/include/CGAL/Surface_mesh/IO/PLY.h#L1019

Should this be done better in both places, e.g. throwing an exception if the vector size does not fit `uchar`?

**Comment by lrineau** [comment]
There should be a test .cpp file using the feature.

**Comment by nh2** [comment]
@lrineau Could you point me at a similar test to get some inspiration from, and how it is added to the test suite? Thanks1

**Comment by afabri** [comment]
Hello @nh2, 
Each package has a subdirectory `test`. And this one has a subdirectory `data`.   So it should go [here](https://github.com/CGAL/cgal/tree/master/Stream_support/test/Stream_support).

File: Stream_support/include/CGAL/IO/PLY/PLY_writer.h (hunk 1 above)
Comment (sloriot) [reply Stream_support/include/CGAL/IO/PLY/PLY_writer.h:302]: There should indeed be a check that `polygon.size()<=std::numeric_limits<unsigned char>::max())`, skipping this property if that's the case.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-22

**Comment by lrineau** [comment]
What is the status of this pull-request. As far as I can see, it seems waiting for tests. See https://github.com/CGAL/cgal/pull/6575#issuecomment-1125875273 and following message.

**Comment by sloriot** [comment]
Exactly, I did some refactoring after the initial patch but the main remaining point is testing the new feature added.

File: Stream_support/include/CGAL/IO/PLY/PLY_writer.h (hunk 1 above)
Comment (sloriot) [reply Stream_support/include/CGAL/IO/PLY/PLY_writer.h:302]: There is actually no "overflow" problem here as the size will simply be modulo `sizeof(uchar)`. Which means that less elements will be put.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-247

**Pull request merged**